src/int_key.rs

```diff
@@ -51,6 +53,28 @@ macro_rules! cw_int_keys {
 
 cw_int_keys!(for i8, u8, i16, u16, i32, u32, i64, u64, i128, u128);
 
+macro_rules! cw_uint_std_keys {
+    (for $($t:ty => $tt:ident),+) => {
+        $(impl IntKey for $t {
+            type Buf = [u8; mem::size_of::<$t>()];
+
+            #[inline]
+            fn to_cw_bytes(&self) -> Self::Buf {
+                self.$tt().to_cw_bytes()
```

Comment:
Can we implement this with

```rust
$tt::from(self)
```

i.e. using the From implementation to get the primitive type. Feels more stable than this somewhat strange getter.

Reply:
This conversion doesn't seem to be present for signed types, probably because the conversion isn't trivial

Reply:
Ah, no, I mean the first one to primitive type. So this in total:

```
$tt::from(self).to_cw_bytes()
```

But it seems like exporters like there is no From for Uint128 -> uint128 and similar. This would be nice to have for sure.

Reply:
Yeah, I did try that. I wanted a quick macro that can cover all the integer types from cosmwasm_std, but I can change it around if you feel it'd be easier for maintenance/understanding

Reply:
I am wondering about a different approach which is more elegant, also looking at 256-bit and 512-bit values for which no primitive type exists.

What `to_cw_bytes`/`from_cw_bytes` does for signed integers is simply flip the first bit of the BE representation (most significant bit).  So instead of going through the primitive type we can just:
- For unsigned integers use `self.to_be_bytes()`
- For signed integers use  `let bytes = self.to_be_bytes()` and the most significant bit of `bytes[0]`

Reply:
You may be right! I'll look at it.

I'd still rather merge this PR without 256 and 512 (seems it would require breaking API, see above). I'll do a follow-up when I'm sure the next release will be breaking anyway.

Reply:
Yeah, I agree with the split. I just think those `iXY()` getter functions to get the primary type are ugly and inconsistent and should not be built upon. However, this should not be a blocker